Source/JavaWrappers.cpp

```diff
@@ -274,6 +281,88 @@ namespace java::lang
     }
 }
 
+namespace java::websocket
+{
+    WebSocketClient::WebSocketClient(std::string url, std::function<void(std::string)> message, std::function<void()> open, std::function<void()> close, std::function<void()> error)
+        : Object{android::global::GetWebSocketClass()}
+        , m_openCallback{std::move(open)}
+        , m_messageCallback{std::move(message)}
+        , m_closeCallback{std::move(close)}
+        , m_errorCallback{std::move(error)}
+    {
+        static JNINativeMethod methods[] = {
+                {"closeCallback", "()V", (void*)OnClose },
+                {"openCallback", "()V", (void*)&WebSocketClient::OnOpen },
+                {"messageCallback", "(Ljava/lang/String;)V", (void*)&WebSocketClient::OnMessage },
+                {"errorCallback", "()V", (void*)&WebSocketClient::OnError },
+        };
+        m_env->RegisterNatives(m_class, methods, 4);
+
+        // initialize socket
+        JObject(m_env->NewObject(m_class, m_env->GetMethodID(m_class, "<init>", "(Ljava/lang/String;)V"), m_env->NewStringUTF(url.c_str())));
+
+        WebSocketClient *p_webSocketClient = this;
+        s_objectMap.insert({JObject(), p_webSocketClient});
+
+        jmethodID connectSocket{m_env->GetMethodID(m_class, "connectBlocking", "()Z")};
+        m_env->CallBooleanMethod(JObject(), connectSocket);
+    }
+
+    void WebSocketClient::OnOpen(JNIEnv* env, jobject obj) {
+        auto itObject = WebSocketClient::FindClientInstance(env, obj);
```

Comment:
Why don't you write a helper function for the map test?

Reply:
`FindClientInstance` is the helper function for finding the instance in the map. Not sure if you mean I should extract something further out into it.

Reply:
I was talking about the code to test whether the object is within the map or not ```itObject != WebSocketClient::s_objectMap.end()```

Reply:
I suggested above that we don't return an iterator. It makes more sense to return the WebSocketClient pointer.

Reply:
Updated to use std::vector instead of std::unordered_map, and also to return the WebSocketClient pointer. I think now, I have to do the `nullptr` check in each one of these callbacks.